pkg/ebpf/c/common/filtering.h

```diff
@@ -190,147 +208,164 @@ statfunc u64 bool_filter_matches(u64 filter_out_scopes, bool val)
+        return 0;
+    }
+
+    u64 res = ~0ULL;
+
     //
-    // boolean filters (not using versioned maps)
+    // boolean filters (not using versioned filter maps)
     //
 
-    // TODO: Create a filter map for each boolean filter (versioning it) #3805
-    if (p->config->cont_filter_enabled_scopes) {
+    if (policies_cfg->cont_filter_enabled_scopes) {
         bool is_container = false;
         u8 state = p->task_info->container_state;
         if (state == CONTAINER_STARTED || state == CONTAINER_EXISTED)
             is_container = true;
-        u64 filter_out_scopes = p->config->cont_filter_out_scopes;
-        u64 mask = ~p->config->cont_filter_enabled_scopes;
+        u64 filter_out_scopes = policies_cfg->cont_filter_out_scopes;
+        u64 mask = ~policies_cfg->cont_filter_enabled_scopes;
+
         // For scopes which has this filter disabled we want to set the matching bits using 'mask'
         res &= bool_filter_matches(filter_out_scopes, is_container) | mask;
     }
 
-    if (p->config->new_cont_filter_enabled_scopes) {
+    if (policies_cfg->new_cont_filter_enabled_scopes) {
         bool is_new_container = false;
         if (p->task_info->container_state == CONTAINER_STARTED)
             is_new_container = true;
-        u64 filter_out_scopes = p->config->new_cont_filter_out_scopes;
-        u64 mask = ~p->config->new_cont_filter_enabled_scopes;
+        u64 filter_out_scopes = policies_cfg->new_cont_filter_out_scopes;
+        u64 mask = ~policies_cfg->new_cont_filter_enabled_scopes;
+
         res &= bool_filter_matches(filter_out_scopes, is_new_container) | mask;
     }
 
-    if (p->config->new_pid_filter_enabled_scopes) {
-        u64 filter_out_scopes = p->config->new_pid_filter_out_scopes;
-        u64 mask = ~p->config->new_pid_filter_enabled_scopes;
+    if (policies_cfg->new_pid_filter_enabled_scopes) {
+        u64 filter_out_scopes = policies_cfg->new_pid_filter_out_scopes;
+        u64 mask = ~policies_cfg->new_pid_filter_enabled_scopes;
+
         res &= bool_filter_matches(filter_out_scopes, proc_info->new_proc) | mask;
     }
 
     //
-    // equality filters (using versioned maps)
+    // equality filters (using versioned filter maps)
     //
 
     u16 version = p->event->context.policies_version;
     void *filter_map = NULL;
 
-    if (p->config->pid_filter_enabled_scopes) {
-        filter_map = get_filter_map(&pid_filter_version, version);
-        u64 filter_out_scopes = p->config->pid_filter_out_scopes;
-        u64 mask = ~p->config->pid_filter_enabled_scopes;
-        u64 max = p->config->pid_max;
-        u64 min = p->config->pid_min;
+    if (policies_cfg->pid_filter_enabled_scopes) {
+        u64 filter_out_scopes = policies_cfg->pid_filter_out_scopes;
+        u64 mask = ~policies_cfg->pid_filter_enabled_scopes;
+        u64 max = policies_cfg->pid_max;
+        u64 min = policies_cfg->pid_min;
 
+        filter_map = get_filter_map(&pid_filter_version, version);
```

Comment:
Shouldn't there be a null check here afterwards for the verifier?

Reply:
That checking is already being done in:

- https://github.com/aquasecurity/tracee/blob/17b2d6464d5c4a6c5d43eeb07bac4a28e0b28460/pkg/ebpf/c/common/filtering.h#L68
- https://github.com/aquasecurity/tracee/blob/17b2d6464d5c4a6c5d43eeb07bac4a28e0b28460/pkg/ebpf/c/common/filtering.h#L92
- https://github.com/aquasecurity/tracee/blob/17b2d6464d5c4a6c5d43eeb07bac4a28e0b28460/pkg/ebpf/c/common/filtering.h#L152